app/validators/course_for_participant_validator.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class CourseForParticipantValidator < ActiveModel::Validator
+  def validate(record)
+    return if record.errors.any?
+    return if has_accepted_application_for_course_given_course_identifier?(record)
+
+    record.errors.add(:course_identifier, :invalid_course)
+  end
+
+private
+
+  def has_accepted_application_for_course_given_course_identifier?(record)
+    return if record.participant.blank?
+
+    record.participant.applications.joins(:course).accepted.active.map { |application|
```

Comment:
I'm unsure about this validation 🤔 do you mind pointing me to where this is in ECF, as I think we just need to validate that the course identifiers exist which I think you're already doing?

Reply:
Hi @cwrw it's [here](https://github.com/DFE-Digital/early-careers-framework/blob/main/app/validators/course_validator.rb#L16) .. it's getting all active participant profiles and then checks the course identifiers.
The idea here is to get all accepted & active applications and do the same with the course identifier?
It's basically checking if the participant has any accepted & active application for the given course..

Reply:
discussed in person: we can tidy the query here and removed active, but need the invalid course validation. Also think about multiple queries doing the same thing and how we can tidy up

Reply:
discussed in person: we're going to amend the query here